src/resizing.jl

```diff
@@ -104,19 +122,42 @@ function imresize(original::AbstractArray{T,N}, new_size::Dims{N}; kwargs...) wh
         if axes(dest) == inds
             copyto!(dest, original)
         else
+            # Non 1-based case as a fallback solution
+            # The OffsetArray case is specially handled to also output OffsetArray
             copyto!(dest, CartesianIndices(axes(dest)), original, CartesianIndices(inds))
         end
     else
         imresize!(similar(original, Tnew, new_size), original; kwargs...)
     end
 end
 
+function imresize(original::OffsetArray{T,N}, new_size::Dims{N}; kwargs...) where {T,N}
+    Tnew = imresize_type(first(original))
+    inds = axes(original)
+    new_inds = map((ax, n)->first(ax):first(ax)+n-1, axes(original), new_size)
+    if map(length, inds) == new_size
+        dest = similar(original, Tnew, new_inds)
+        if axes(dest) == new_inds
+            # a trivial case of OffsetArray
+            copyto!(parent(dest), original)
+        else
+            copyto!(dest, CartesianIndices(axes(dest)), original, CartesianIndices(inds))
+        end
+    else
+        dest = imresize(original, new_inds; kwargs...)
+    end
+    return dest
+end
+
 function imresize(original::AbstractArray{T,N}, new_inds::Indices{N}; kwargs...) where {T,N}
     Tnew = imresize_type(first(original))
+    # The pirated `similar` method from OffsetArrays will be triggered
+    # thus for type stability, we unconditionally output an `OffsetArray`.
+    origin = OffsetArrays.Origin(map(first, new_inds))
     if axes(original) == new_inds
-        copyto!(similar(original, Tnew), original)
+        OffsetArray(copyto!(similar(original, Tnew), original), origin)
```

Comment:
Rather than manually create an `OffsetArray`, how about use `similar(original, Tnew, new_inds)`? That will create an OffsetArray if `new_inds[i] isa UnitRange` rather than a `Base.OneTo`.